### PR TITLE
[Bug] Fix url redirection, add default render path

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
1. Fix url redirection
2. Add default render path when there is basePath
    closing : https://github.com/opensearch-project/dashboards-query-workbench/pull/381
(Support for dev tools under modal pop out)
As we will make dev tools as a full page modal when new nav toggle is turned on, we will use MemoryRouter to replace HashRouter in our page to avoid modifying the hosted page' hash. And this change will impact search relevance functionality.
And this PR adds a default path when there is no '/' route.
For more details please take a look on the comment: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7938#discussion_r1738383931.

Tested url paths:
path={`/${basePath}/:dataSource`}
path={`/${basePath}/accelerate/:dataSource`}
path="/"
path={`/${basePath}`}

Old Navigation:

https://github.com/user-attachments/assets/128c154d-e4be-4ab8-ba19-3f974306b010

New Navigation:

https://github.com/user-attachments/assets/5b225238-e4e9-4181-a020-58b89981261a

New Navigation + https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7938 modal applied.

https://github.com/user-attachments/assets/592fd5ff-78c7-4e3f-91f5-349fe3aaf14a

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/app.tsx [hunk 1]
```diff
@@ -108,6 +108,27 @@ export const WorkbenchApp = ({
                     />
                   )}
                 />
+                {basePath && (
```
Comment (ps48) [inline public/components/app.tsx:111]: can this be changed to `{isNavGroupEnabled && (` ?

Comment (ps48) [reply public/components/app.tsx:111]: Also, I think we can merge this with route on line 54. May be changing the line `#56` to 
```
path=["/",{isNavGroupEnabled ? basePath : `/${basePath}`}]
```

File: public/components/Main/main.tsx [hunk 2]
```diff
@@ -291,7 +291,7 @@ export class Main extends React.Component<MainProps, MainState> {
       refreshTree: false,
       isAccelerationFlyoutOpened: false,
       isCallOutVisible: false,
-      cluster: 'Indexes',
+      cluster: props.urlDataSource?'Data source Connections':'Indexes',
```
Comment (ps48) [inline public/components/Main/main.tsx:294]: Need formatting fix with spacing here.

File: public/components/app.tsx (hunk 1 above)
Comment (TackAdam) [reply public/components/app.tsx]: Yes, makes it more clear as well good call out.

File: public/components/Main/main.tsx (hunk 2 above)
Comment (TackAdam) [reply public/components/Main/main.tsx]: Changed too
      cluster: props.urlDataSource ? 'Data source Connections' : 'Indexes',

File: public/components/app.tsx [hunk 3]
```diff
@@ -108,6 +108,27 @@ export const WorkbenchApp = ({
...
+                        notifications={notifications}
+                        savedObjects={savedObjects}
+                        dataSourceEnabled={dataSourceEnabled}
+                        dataSourceManagement={dataSourceManagement}
+                        dataSourceMDSId={dataSourceId}
+                        setActionMenu={setActionMenu}
+                      />
+                    )}
+                  />
+                )}
```
Comment (ps48) [inline public/components/app.tsx:131]: Any luck in removing this and merging it with route of line 54?

Comment (ps48) [reply public/components/app.tsx:131]: With `path={isNavGroupEnabled ? ["/", basePath] : `/${basePath}`}`

Comment (TackAdam) [reply public/components/app.tsx]: Updated path to 
`path={isNavGroupEnabled ? [`${basePath}`, `/`] : `/${basePath}`}`
Tested and seems good.

File: public/components/app.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ export const WorkbenchApp = ({
               <Switch>
                 <Route
                   exact
-                  path={`/${basePath}`}
+                  path={isNavGroupEnabled ? [`${basePath}`, `/`] : `/${basePath}`}
```
Comment (SuZhou-Joe) [inline public/components/app.tsx:56]:
```suggestion
                  path={isNavGroupEnabled ? [`${basePath}`, `/`] : `${basePath}`}
```

Comment (TackAdam) [reply public/components/app.tsx:56]: For the url re-direction to work correctly while the new home page is disabled the "/" is required or the following fail:

path="/"
path={/${basePath}}

Comment (SuZhou-Joe) [reply public/components/app.tsx:56]: I see, yeah you are right.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Comment by ps48** [comment]
Thanks @TackAdam for the bug fix 🥳

**Pull request merged**